### ISSUE TITLE
[BUG/Help] What is the loss function in training and fine-tuning the model?

### ISSUE THREAD

**Issue by 529106896** [THUDM/ChatGLM-6B#936]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I tried using soft labels as a response when fine-tuning the model, but I found that the generated model only approached the response in terms of text similarity and did not actually use soft labels, resulting in local optima. I tried to see what the loss function was in the code, but I couldn't find it. I guess the loss function should use text similar differences between texts, such as BLEU or ROUGE scoring?

我在进行模型微调时尝试过使用软标签作为response，但我发现生成的模型只是在文本相似度上去接近response，并没有真正使用到软标签，因此导致了局部最优。我尝试在代码中查看损失函数到底是什么，但并没有找到。我猜损失函数应该是使用文本之间的相似差异，比如使用BLEU或者ROUGE评分？

### Expected Behavior

I only found these codes in trainer.py, but I don't know what the loss function is, or how to calculate loss.
```python
# tr_loss is a tensor to avoid synchronization of TPUs through .item()
tr_loss = torch.tensor(0.0).to(args.device)
# _total_loss_scalar is updated everytime .item() has to be called on tr_loss and stores the sum of all losses
self._total_loss_scalar = 0.0
```

### Steps To Reproduce

1. Open trainer.py in ./ptuning/trainer.py
2. Find these codes from line 1813 to 1818

### Environment

```markdown
- OS: Centos
- Python: 3.10
- Transformers: 4.26.1
- PyTorch: 1.12
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by zplvzrm** [comment]
https://huggingface.co/THUDM/chatglm-6b/blob/main/modeling_chatglm.py
